packages/xo-server/src/api/host.mjs

```diff
@@ -456,3 +456,52 @@ setControlDomainMemory.params = {
 setControlDomainMemory.resolve = {
   host: ['id', 'host', 'administrate'],
 }
+
+// -------------------------------------------------------------------
+/**
+ *
+ * @param {{host:HOST}} params
+ * @returns null if plugin is not installed or don't have the method
+ *          an object device: status on success
+ */
+export function getSmartctlHealth({ host }) {
+  return this.getXapi(host).getSmartctlHealth(host._xapiId)
+}
+
+getSmartctlHealth.description = 'get smartctl health status'
+
+getSmartctlHealth.params = {
+  id: { type: 'string' },
+}
+
+getSmartctlHealth.resolve = {
+  host: ['id', 'host', 'administrate'],
+}
+
+/**
+ *
+ * @param {{host:HOST}} params
+ * @returns null if plugin is not installed or don't have the method
+ *          an object device: full device information on success
+ */
+export function getSmartctlInformation({ host, deviceNames }) {
+  return this.getXapi(host).getSmartctlInformation(host._xapiId, deviceNames)
+}
+
+getSmartctlInformation.description = 'get smartctl information'
+
+getSmartctlInformation.params = {
+  id: { type: 'string' },
+
+  deviceNames: {
+    type: 'array',
+    items: {
+      type: 'string',
+    },
+    optional: true,
+  },
+}
+
+getSmartctlInformation.resolve = {
+  host: ['id', 'host', 'administrate'],
```

Comment:
Is this necessary? What is the user is only a viewer or operator? Will it fail?

Reply:
I am not sure, how can I check this  ? it took this from the hyper threading plugin

Reply:
IMO, if the call doesn't alter the host, `'view'` is enough.